web/pdf_page_view.js

```diff
@@ -320,32 +356,36 @@ class PDFPageView {
     keepAnnotationLayer = false,
     keepAnnotationEditorLayer = false,
     keepXfaLayer = false,
+    keepTextLayer = false,
   } = {}) {
     this.cancelRendering({
       keepAnnotationLayer,
       keepAnnotationEditorLayer,
       keepXfaLayer,
+      keepTextLayer,
     });
     this.renderingState = RenderingStates.INITIAL;
 
     const div = this.div;
-    div.style.width = Math.floor(this.viewport.width) + "px";
-    div.style.height = Math.floor(this.viewport.height) + "px";
+    div.style.width = Math.round(this.viewport.width) + "px";
+    div.style.height = Math.round(this.viewport.height) + "px";
```

Comment:
Isn't there a potential risk that this div-container (i.e. the `page`) may end up at a different size than the `canvasWrapper` div, which could thus have unforeseen side-effects?

In any case, this is now also inconsistent with the existing code in https://github.com/mozilla/pdf.js/blob/4f1b6f345bc83fd1aca6eef3a3338a480e01251b/web/pdf_page_view.js#L177-L178

Reply:
The `canvasWrapper` has the same dimensions as the `page`:
https://github.com/mozilla/pdf.js/blob/4f1b6f345bc83fd1aca6eef3a3338a480e01251b/web/pdf_page_view.js#L685

Reply:
OK, that's good; thank you!

However is there a risk that the page/canvasWrapper now becomes *larger* than the `style` of the canvas-element itself, or does the following code guarantee that won't happen?
https://github.com/mozilla/pdf.js/blob/4f1b6f345bc83fd1aca6eef3a3338a480e01251b/web/pdf_page_view.js#L952-L955

Reply:
I had a look on this code and if there's something wrong because of a dimension mismatch it's a pre-existing issue.
See with the Firefox viewer on a screen with a deviceRatio equals to 1:
![image](https://user-images.githubusercontent.com/5641725/204010570-a44cb6c5-debe-4c52-bdc9-21ebc4530de9.png)
Anyway as far as I understand this code the goal is to have some dimensions which avoid decimal dimensions when the system itself applies the deviceRatio. So it doesn't seems to be a problem for now and it shouldn't be one in the future it's likely an "inconsistency" we'll have to fix at some point.